api/app.py

```diff
@@ -42,10 +42,11 @@ def create_app(config_object=None):
         handlers.register(app)
         register_specs(app)
 
-        app.config['HUB_URL'] = urljoin(
-            app.config['SERVICE_URL'],
-            url_for('views.subscriptions_by_id', _external=False)
-        )
+        app.config['HUB_URL'] = f"{app.config['SERVICE_URL']}/messages/subscriptions/by_id"
+        # app.config['HUB_URL'] = urljoin(
```

Comment:
@magicjohnson this wasn't working for me:
```
api_1                | Traceback (most recent call last):
api_1                |   File "manage.py", line 9, in <module>
api_1                |     app = create_app(config_object=DevelopmentConfig())
api_1                |   File "/src/api/app.py", line 48, in create_app
api_1                |     url_for('views.subscriptions_by_id', _external=False)
api_1                |   File "/usr/local/lib/python3.6/site-packages/flask/helpers.py", line 333, in url_for
api_1                |     "Application was not able to create a URL adapter for request"
api_1                | RuntimeError: Application was not able to create a URL adapter for request independent URL generation. You might be able to fix this by setting the SERVER_NAME config variable.
```

I had a bit of a pain doing the rebase due to using a new tool I haven't used much (SublimeMerge) and might have broken something, but I'm not sure. I don't think we want to have server name if possible, because it gives us 404s if the domain name doesn't match exactly. `localhost:8180` should still be valid!

Reply:
I checked test suite, it was fine, should have checked server as well... sorry for that

this should work:
```
        with app.test_request_context():
            app.config['HUB_URL'] = urljoin(
                app.config['SERVICE_URL'],
                url_for('views.subscriptions_by_id', _external=False)
            )
```

P.S. some flask things are hacky...so I'm fine with current version to concat SERVICE_URL and url

Reply:
Yeah, I don't like the `test` part in `test_request_context`, but prefer to not hardcode url paths, so going with your suggestion. Thanks @magicjohnson !